hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -1713,37 +1713,39 @@ type Chunk chunks dim shape dtype device = ChunkCheck shape dim (ChunkImpl (Chun
 
 -- | chunk
 --
--- >>> :type chunk @3 @1 (ones :: CPUTensor 'D.Float '[2, 2])
--- chunk @3 @1 (ones :: CPUTensor 'D.Float '[2, 2])
---   :: HList
---        '[Tensor '( 'D.CPU, 0) 'D.Float '[2, 1],
---          Tensor '( 'D.CPU, 0) 'D.Float '[2, 1]]
+-- -- >>> :type chunk @3 @1 (ones :: CPUTensor 'D.Float '[2, 2])
+-- -- chunk @3 @1 (ones :: CPUTensor 'D.Float '[2, 2])
+-- --   :: HList
+-- --        '[Tensor '( 'D.CPU, 0) 'D.Float '[2, 1],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[2, 1]]
 -- >>> t0 :. t1 :. HNil = chunk @3 @1 (ones :: CPUTensor 'D.Float '[2, 2])
 -- >>> dtype &&& shape $ t0
 -- (Float,[2,1])
 -- >>> dtype &&& shape $ t1
 -- (Float,[2,1])
--- >>> :type chunk @3 @1 (ones :: CPUTensor 'D.Float '[1, 0, 3])
--- chunk @3 @1 (ones :: CPUTensor 'D.Float '[1, 0, 3])
---   :: HList
---        '[Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3],
---          Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3],
---          Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3]]
+--
+-- -- >>> :type chunk @3 @1 (ones :: CPUTensor 'D.Float '[1, 0, 3])
+-- -- chunk @3 @1 (ones :: CPUTensor 'D.Float '[1, 0, 3])
+-- --   :: HList
+-- --        '[Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[1, 0, 3]]
 -- >>> t0 :. t1 :. t2 :. HNil = chunk @3 @1 (ones :: CPUTensor 'D.Float '[1, 0, 3])
 -- >>> dtype &&& shape $ t0
 -- (Float,[1,0,3])
 -- >>> dtype &&& shape $ t1
 -- (Float,[1,0,3])
 -- >>> dtype &&& shape $ t2
 -- (Float,[1,0,3])
--- >>> :type chunk @6 @0 (ones :: CPUTensor 'D.Float '[19, 4])
--- chunk @6 @0 (ones :: CPUTensor 'D.Float '[19, 4])
---   :: HList
---        '[Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
---          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
---          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
---          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
---          Tensor '( 'D.CPU, 0) 'D.Float '[3, 4]]
+--
+-- -- >>> :type chunk @6 @0 (ones :: CPUTensor 'D.Float '[19, 4])
+-- -- chunk @6 @0 (ones :: CPUTensor 'D.Float '[19, 4])
+-- --   :: HList
+-- --        '[Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[4, 4],
+-- --          Tensor '( 'D.CPU, 0) 'D.Float '[3, 4]]
```

Comment:
Nice!

Reply:
well, it was nice before. for some reason, ghci refuses to expand these types now (I'm not sure what has changed that would cause that), and the doctests here failed. that's why I commented them for now.